src/pages/index.js

```diff
@@ -15,86 +16,215 @@ import {
   addNewCardButton,
   profileTitleInput,
   profileDescriptionInput,
+  editAvatarButton,
+  avatarEditFormElement,
 } from "../utils/constants.js";
 
+// Api
+const api = new Api({
+  baseUrl: "https://around-api.en.tripleten-services.com/v1",
+  headers: {
+    authorization: "c00d25e4-e498-4e3f-a0a9-735cb37a7bb4",
+    "Content-Type": "application/json",
+  },
+});
+
 //validators
 const editProfileFormValidator = new FormValidator(config, profileFormElement);
 editProfileFormValidator.enableValidation();
 
 const addCardValidator = new FormValidator(config, addCardFormElement);
 addCardValidator.enableValidation();
 
-//new card element
+const avatarFormValidator = new FormValidator(config, avatarEditFormElement);
+avatarFormValidator.enableValidation();
+
+//CARD FUNCTIONS
+
+//get initial cards
+let section;
+api
+  .getInitialCards()
+  .then((cards) => {
+    section = new Section(
+      {
+        items: cards,
+        renderer: (cardData) => {
+          const cardElement = createCard(cardData);
+          section.addItem(cardElement);
+        },
+      },
+      ".cards__list"
+    );
+
+    section.renderItems();
+  })
+  .catch((err) => console.error(err));
+
+//like and dislike a card
+function handleLikeClick(card) {
+  if (card._isLiked) {
+    api
+      .dislikeCard(card._id)
+      .then(() => {
+        card.toggleLike();
+        card._isLiked = false;
+      })
+      .catch((err) => {
+        console.error(`Error on Card Dislike ${err}`);
+      });
+  } else {
+    api
+      .likeCard(card._id)
+      .then(() => {
+        card.toggleLike();
+        card._isLiked = true;
+      })
+      .catch((err) => {
+        console.error(`Error on Card Like ${err}`);
+      });
+  }
+}
+
+//add card functionality
 const createCard = (data) => {
-  const card = new Card(data, "#card-template", handleImagePreview);
+  const card = new Card(
+    data,
+    "#card-template",
+    handleImagePreview,
+    handleDeleteCard,
+    handleLikeClick
+  );
   return card.getCardElement();
 };
 
-//render card elements
-const cardListEl = new Section(
-  {
-    items: initialCards,
-    renderer: (data) => {
-      cardListEl.addItem(createCard(data));
-    },
-  },
-
-  ".cards__list"
+const addCardPopup = new PopupWithForm(
+  "#add-card-modal",
+  handleAddCardFormSubmit
 );
+addCardPopup.setEventListeners();
+
+addNewCardButton.addEventListener("click", () => {
+  addCardPopup.open();
+});
 
-cardListEl.renderItems();
+function handleAddCardFormSubmit(data) {
+  const name = data.title.trim();
+  const link = data.link.trim();
+  addCardPopup.close();
+  addCardFormElement.reset();
+  addCardValidator.disableSubmitButton();
```

Comment:
You need to **disable** the button only in a `then` block after a **successful response** to be able to click it again if there is a server error.

Reply:
**Please, correct it everywhere in the code**